### PR TITLE
Object player/1358 fix broken links in cadence document

### LINKED ISSUE #1358: Fix broken links in Cadence documentation
### Problem
Many links in Cadence documentation are broken. We should do a full audit to address as many as we can

### Steps to Reproduce
check out https://docs.onflow.org/cadence/language/

### PR DESCRIPTION
**Pull request by ObjectPlayer**
Closes #1358

## Description

Resolved all broken links in the cadence documents.
______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1099,7 +1099,9 @@ For `AuthAccount`:
   - `monaco-languageclient-cadence`: Language Server Protocol client for the the Monaco editor
   - `cadence-parser`: The Cadence parser
 
-  In addition, there are also examples for the [language server](https://github.com/onflow/cadence/tree/master/npm-packages/cadence-language-server-demo) and the [parser](https://github.com/onflow/cadence/tree/master/npm-packages/cadence-parser-demo) that demonstrate the use of the packages.
+  In addition, there is also an example for the [language server](https://github.com/onflow/cadence/tree/master/npm-packages/cadence-language-server-demo) 
+  <!-- and the [parser](https://github.com/onflow/cadence/tree/master/npm-packages/cadence-parser-demo) that demonstrate the use of the packages.
+  >>
```
Comment (turbolent) [inline CHANGELOG.md:1104]:
```suggestion
  In addition, there is also [an example for the language server](https://github.com/onflow/cadence/tree/master/npm-packages/cadence-language-server-demo), and an [AST exploration tool](https://github.com/onflow/cadence/tree/master/tools/astexplorer) that demonstrate the use of the parser package.
```

File: docs/language/access-control.md [hunk 2]
```diff
@@ -99,7 +99,7 @@ To summarize the behavior for functions:
 | `access(account)`        | Current, inner, and other contracts in same account |
 | `pub` / `access(all)`    | **All**                                             |
 
-Declarations of structures, resources, events, and [contracts](../contracts) can only be public.
+Declarations of structures, resources, events, and [contracts](https://github.com/onflow/cadence/blob/master/docs/language/contracts.md) can only be public.
```
Comment (turbolent) [inline docs/language/access-control.md:102]: The relative links are correct here, the links should not link to the source files: The documentation source files in this repository are rendered on https://docs.onflow.org/cadence/language.

Could you please revert this and the other similar link changes below?

File: encoding/README.md [hunk 3]
```diff
@@ -2,6 +2,6 @@
 
 The `encoding` packages contain functions to encode and decode Cadence values to other formats.
 
-Currently, only the [JSON-Cadence](https://github.com/onflow/flow/blob/master/docs/json-cadence-spec.md) format is supported.
+Currently, only the [JSON-Cadence](https://github.com/onflow/cadence/blob/master/docs/json-cadence-spec.md) format is supported.
```
Comment (turbolent) [inline encoding/README.md:5]: This link should actually point to the rendered documentation page on the website:

```suggestion
Currently, only the [JSON-Cadence](https://docs.onflow.org/cadence/json-cadence-spec/) format is supported.
```

**Review by turbolent — COMMENTED** [review]
Thank you for the suggestions!

File: CHANGELOG.md (hunk 1 above)
Comment (ObjectPlayer) [reply CHANGELOG.md:1104]: there is no parser-demo in the code, that's why it causes a broken-link issue, therefore I commented parser link. AST exploration is also a good choice to use here instead.

File: docs/language/access-control.md (hunk 2 above)
Comment (ObjectPlayer) [reply docs/language/access-control.md:102]: Actually, all the relative links that I have updated are not working properly, which move to that link it showing you "404, page not found" error, therefore I used absolute path from the source.
Let me revert all these links according to https://docs.onflow.org/cadence/language

File: encoding/README.md (hunk 3 above)
Comment (ObjectPlayer) [reply encoding/README.md:5]: Okey, I get the point let me update the code accordingly. Thanks for responding

File: docs/language/access-control.md (hunk 2 above)
Comment (ObjectPlayer) [reply docs/language/access-control.md]: Now I get that, the path in the github repository is fine but when you move to https://docs.onflow.org/cadence/language you can see both relative links at the end of the page or not working 

> Disclaimer: In real Cadence code, all type definitions and code must be declared and contained in [contracts](https://docs.onflow.org/cadence/contracts/) or [transactions](https://docs.onflow.org/cadence/transactions/), but we omit these containers in examples for simplicity.

let me fix these links

Comment (kerrywei) [reply docs/language/access-control.md]: @ObjectPlayer we should hold on converting to links pointing to flow docs site, as the relative linking issue has been addressed by PR https://github.com/onflow/flow/pull/764/

Comment (turbolent) [reply docs/language/access-control.md]: @kerrywei fixed the relative links in #1372, I updated this PR (merged master). Relative links will also help us build and test the documentation locally. I left the other two improvements in the PR, as they are good fixes. Thank you again for raising this issue and helping to fix it @ObjectPlayer 🙏

**Review by turbolent — APPROVED** [review]

**Pull request merged**